gravitee-apim-console-webui/src/management/api/notifications/notifications-add-notification-dialog/notifications-add-notification-dialog.component.ts

```diff
@@ -13,11 +13,39 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Component } from '@angular/core';
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { Notifier } from '../../../../entities/notification/notifier';
+import { NewNotification } from '../../../../entities/notification/newNotification';
+
+export interface NotificationsAddNotificationDialogData {
+  notifier: Notifier[];
+}
+export type NotificationsAddNotificationDialogResult = NewNotification;
 
 @Component({
   selector: 'notifications-add-notification-dialog',
   template: require('./notifications-add-notification-dialog.component.html'),
   styles: [require('./notifications-add-notification-dialog.component.scss')],
 })
-export class NotificationsAddNotificationDialogComponent {}
+export class NotificationsAddNotificationDialogComponent {
+  notificationForm: FormGroup;
+  notifierList: Notifier[];
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) dialogData: NotificationsAddNotificationDialogData,
+    public dialogRef: MatDialogRef<NotificationsAddNotificationDialogData, NotificationsAddNotificationDialogResult>,
+  ) {
+    this.notifierList = dialogData.notifier;
+    this.notificationForm = new FormGroup({
+      name: new FormControl(null, [Validators.required]),
+      notifier: new FormControl(this.notifierList[0].id, [Validators.required]),
+    });
+  }
+
+  onSubmit() {
+    this.dialogRef.close(this.notificationForm.getRawValue());
```

Comment:
here I think part of the object is missing. 
what do you think about creating a more robust object like 
```
const {name, notifier} = (this.notificationForm.getRawValue()
this.dialogRef.close({
name,
notifier,
.. missing par of NotificationSettings
})
```